### PR TITLE
Added x-ms-examples for documentdb database account operations

### PR DESCRIPTION
**Pull request by dmakwana**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [X] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountCheckNameExists.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "parameters": {
+    "accountName": "ddb1",
+    "api-version": "2015-04-08",
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountCheckNameExists.json]: remove the extra comma

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountCreateMax.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
+  "parameters": {
+    "accountName": "ddb1",
+    "resourceGroupName": "rg1",
+    "api-version": "2015-04-08",
+    "subscriptionId": "subid",
+    "location": "westus",
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountCreateMax.json]: general comment: please add the body parameter under the name of the body parameter defined in swagger.
```json5 
"createUpdateParameters": { . . .  entire body parameter here...}
```

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountListKeys.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    "api-version": "2015-04-08",
+    "subscriptionId": "subid"
+  },
+  "responses": {
+    "200": { 
+      "body": {
+        "primaryMasterKey": "primaryMasterKey",
+        "secondaryMasterKey": "secondaryMasterKey",
+        "primaryReadonlyMasterKey": "primaryReadonlyMasterKey",
+        "secondaryReadonlyMasterKey": "secondaryReadonlyMasterKey"
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountListKeys.json]: General comment: please don't flatten properties. Flattening happens at the sdk level. On the wire things are the same as mentioned in the swagger spec.
So this should be:
```json5
"body": {
        "primaryMasterKey": "primaryMasterKey",
        "secondaryMasterKey": "secondaryMasterKey",
        "properties": {
          "primaryReadonlyMasterKey": "primaryReadonlyMasterKey",
          "secondaryReadonlyMasterKey": "secondaryReadonlyMasterKey"
        }
      }
```

**Review by amarzavery — CHANGES_REQUESTED** [review]
Have made general comments. Please make sure that similar changes are made everywhere

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountPatch.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{
+    "parameters": {
+      "accountName": "ddb1",
+      "resourceGroupName": "rg1",
+      "api-version": "2015-04-08",
+      "subscriptionId": "subid",
+      "updateParameters": {
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountPatch.json]: tags should be outside of properties.
```json5
"updateParameters": {
  "tags": {
     "dept": "finance"
   }
}
```

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountCreateMax.json [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  "parameters": {
+    "accountName": "ddb1",
+    "resourceGroupName": "rg1",
+    "api-version": "2015-04-08",
+    "subscriptionId": "subid",
+    "createUpdateParameters": {
+      "location": "westus",
+      "tags": {},
+      "properties": {
+        "kind": "GlobalDocumentDB",
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountCreateMax.json]: kind should outside of the properties object. It should one level up along with tags and location.

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountFailoverPriorityChange.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "parameters": {
+    "accountName": "ddb1-failover",
+    "resourceGroupName": "rg1",
+    "api-version": "2015-04-08",
+    "subscriptionId": "subid",
+    "failoverPolicies": [
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountFailoverPriorityChange.json]: the name of the parameter is "failoverParameters". So the example should be like this:

```json5
"failoverParameters": {
  "failoverPolicies": [
       {
         "locationName": "eastus",
         "failoverPriority": 0
       },
       {
         "locationName": "westus",
         "failoverPriority": 1
       }
   ]
}
```

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountRegenerateKey.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
+  "parameters": {
+    "accountName": "ddb1",
+    "resourceGroupName": "rg1",
+    "api-version": "2015-04-08",
+    "subscriptionId": "subid",
+    "keyToRegenerate": {
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountRegenerateKey.json]: The example should be like this, based on the swagger spec:
```json5
"keyToRegenerate": {
        "keyKind": "primary"
    }
```

File: arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountRegenerateKey.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+      "properties": {
+        "keyKind": "primary"
+      }
+    }
+  },
+  "responses": {
+    "202": { }
```
Comment (amarzavery) [inline arm-documentdb/2015-04-08/examples/DocumentDBDatabaseAccountRegenerateKey.json]: missing example for 200.
```
"200": {}
```

**Review by amarzavery — CHANGES_REQUESTED** [review]

**Comment by dmakwana** [comment]
Unsure as to why non-documentdb related changes that already exist in the master branch are showing up as changed files in this PR. I can close and create another PR

**Comment by fearthecowboy** [comment]
@dmkwana we're seeing that in a few places in github. Ignore that for now

**Comment by amarzavery** [comment]
@dmakwana - The model validator is integrated into Travis CI. It failed in the CI. (The build job is green because we are allowing failures (we are doing a dry run right now)).
Please check this [link](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/199001504#L2348) and fix the examples accordingly. Let me know if the error message needs any clarification.

**Comment by amarzavery** [comment]
@vishrutshah @sarangan12 @salameer - The linter is showing a false positive over [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/199001500#L2344).

```
WARNING: TrackedResourceValidation - Tracked Resource failing validation is: DatabaseAccountCreateUpdateParameters. Validation Failed: 1. 
    A Tracked Resource must have: 
    1. A Get Operation 
    2. A ListByResourceGroup operation with x-ms-pageable extension and 
    3. A ListBySubscriptionId operation with x-ms-pageable extension.
    4. Type, Location, Tags should not be used in the properties.
	Path: #/Definitions
```

[DatabaseAccountCreateParameters](https://github.com/dmakwana/azure-rest-api-specs/blob/2a2d72c4ebdc60b42128a0c5f2aed8fd6541368e/arm-documentdb/2015-04-08/swagger/documentdb.json#L659) is used as an input to the PUT operation. It is an x-ms-azure-resource (basically a tracked resource). However, for some reason the service cannot send and get back the same model in the PUT. Therefore they have two separate models. DatabaseAccountCreateParameters is the input and DatabaseAccount is the output. The linter should verify that the output type must satisfy the above 4  conditions. Not the intput type (as it is not really a resource).

**Review by amarzavery — APPROVED** [review]

**Pull request merged**